File: Creational/FactoryMethod/TypeScript/Structure/ConcreteProductA.ts
import { Product } from "./Product";

/**
 * ConcreteProductA - Concrete implementation of Product
 * Represents a specific type of product created by ConcreteCreatorA
 */
export class ConcreteProductA extends Product {
  operation(): string {
    return "ConcreteProductA operation";
  }
}

File: Creational/FactoryMethod/Java/Structure/ConcreteProductA.java
/**
 * ConcreteProductA - implementation cụ thể của Product
 */
public class ConcreteProductA implements Product {
    
    @Override
    public void operation() {
        System.out.println("ConcreteProductA đang thực hiện thao tác");
    }
    
    @Override
    public String getName() {
        return "Sản phẩm A";
    }
}
